### PR TITLE
Provide a usefull error when ask*() is not used wihtin a task()

### PR DESCRIPTION
**Pull request by staabm**
| Q             | A
| ------------- | ---
| Bug fix?      | No
| New feature?  | Yes
| BC breaks?    | No
| Deprecations? | No
| Fixed tickets | N/A

before this change a newcomer to this project sees errors like 
```
PHP Fatal error:  Uncaught Error: Call to a member function getOutput() on boolean in /cluster/www/www/www/tools/deployer/src/functions.php:677
Stack trace:
#0 /cluster/www/www/www/tools/deployer/src/functions.php(685): Deployer\output()
#1 /cluster/www/www/www/tools/deployer/src/functions.php(567): Deployer\isQuiet()
#2 /deploy/kunzmann/stage/rocket/deploy.php(51): Deployer\ask('Zielstage (prod...')
#3 /cluster/www/www/www/tools/deployer/bin/dep(106): require('/deploy/kunzman...')
#4 /cluster/www/www/www/tools/deployer/bin/dep(107): {closure}()
#5 {main}
  thrown in /cluster/www/www/www/tools/deployer/src/functions.php on line 677
```

when using a `ask()` function within global scope instead of a `task()`.

After this change the error turns into

```
PHP Fatal error:  Uncaught Deployer\Exception\Exception: 'Deployer\ask' can only be used within a 'task()'-function! in /cluster/www/www/www/tools/deployer/src/functions.php:757
Stack trace:
#0 /cluster/www/www/www/tools/deployer/src/functions.php(567): Deployer\requiresTaskContext('Deployer\\ask')
#1 /deploy/kunzmann/stage/rocket/deploy.php(51): Deployer\ask('Zielstage (prod...')
#2 /cluster/www/www/www/tools/deployer/bin/dep(106): require('/deploy/kunzman...')
#3 /cluster/www/www/www/tools/deployer/bin/dep(107): {closure}()
#4 {main}
  thrown in /cluster/www/www/www/tools/deployer/src/functions.php on line 757
```

based on a "discussion" in https://github.com/deployphp/deployer/issues/371#issuecomment-285592637

### CODE REVIEW

**Comment by staabm** [comment]
This small new function can obviously be used in more places but I thought this would be a good start.

**Comment by elfet** [comment]
Cool.  i think this is helpful.

**Comment by elfet** [comment]
Please update changelog.

**Comment by staabm** [comment]
done.

File: src/functions.php [hunk 1]
```diff
@@ -742,3 +751,16 @@ function parse($value)
...
     return Context::get()->getEnvironment()->parse($value);
 }
+
+/**
+ * Throws a Exception when not called within a task-context.
+ *
+ * This method provides a usefull error to the end-user to make him/her aware
+ * to use a function in the required task-context.
+ */
+function requiresTaskContext($callerName)
```
Comment (elfet) [inline src/functions.php]: I think better to create method in `Context`:

```php
Context::required();
```

Comment (staabm) [inline src/functions.php]: also thought about that option. will change it.

Comment (staabm) [inline src/functions.php]: now I know why I decided against a method in Context ;-):

the message here is rather task specific and `Context` class atm is rather generic.

```
    /**
     * Throws a Exception when not called within a task-context.
     *
     * This method provides a usefull error to the end-user to make him/her aware
     * to use a function in the required task-context.
     *
     * @throws Exception
     * @return Context
     */
    public static function required($callerName)
    {
        $ctx = self::get();
        if (!$ctx) {
            throw new Exception("'$callerName' can only be used within a 'task()'-function!");
        }
        return $ctx;
    }
```

do you still think we should move it to Context, I am fine with any solution you will accept.

Comment (elfet) [inline src/functions.php]: I think its better.

**Comment by staabm** [comment]
adjusted per feedback

**Pull request merged**